cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -363,9 +381,23 @@ func (ks *kube2sky) removeService(obj interface{}) {
 }
 
 func (ks *kube2sky) updateService(oldObj, newObj interface{}) {
-	// TODO: Avoid unwanted updates.
-	ks.removeService(oldObj)
-	ks.newService(newObj)
+	oldSvc, oldOk := oldObj.(*kapi.Service)
+	newSvc, newOk := newObj.(*kapi.Service)
+	if oldOk && newOk {
+		var (
+			oldRec recordTree
+			newRec recordTree
+		)
+		ks.addServiceRecords(&oldRec, oldSvc)
+		ks.addServiceRecords(&newRec, newSvc)
+		toSet, toRemove := recordDiff(&oldRec, &newRec)
+		ks.mutateEtcdOrDie(func() error { return toRemove.RemoveEtcd(ks.etcdClient) })
+		ks.mutateEtcdOrDie(func() error { return toSet.SetEtcd(ks.etcdClient) })
+	} else if oldOk {
```

Comment:
Are you sure this is what we see when a Service is deleted?  Not some neutered-but-valid Service object?